pylxd/client.py

```diff
@@ -130,6 +130,8 @@ def __init__(self, endpoint=None, version='1.0', cert=None, verify=True):
             auth = response.json()['metadata']['auth']
             if auth != "trusted":
                 raise exceptions.ClientAuthenticationFailed()
+
+            self.host_info = response.json()['metadata']['environment']
```

Comment:
I would cache the entire `metadata` info itself. There are all sorts of things in that dict that are good for us to know for various reasons.